File: backend/ProgrammingInternshipPlatform/ProgrammingInternshipPlatform.Application/Abstractions/GraphApi/Responses/Account.cs
using ProgrammingInternshipPlatform.Domain.AccountManagement.Identifiers;

namespace ProgrammingInternshipPlatform.Application.Abstractions.GraphApi.Responses;

public class Account
{
    public AccountId Id { get; init; }
    public string DisplayName { get; init; }
    public string GivenName { get; init; }
    public string Surname { get; init; }
    public string Initials { get; init; }
    public string JobTitle { get; init; }
    public string Email { get; init; }
}

File: shine/src/app/shared/models/account.ts
export interface Account {
    coordinatorId: string;
    displayName: string;
    givenName: string;
    surname: string;
    initials: string;
    jobTitle: string;
    email: string;
    shortenedEmail: string;
    role: string;
}

export interface AccountWithPicture extends Account {
    pictureUrl?: string;
}